library/ssl_tls13_generic.c

```diff
@@ -1454,6 +1454,47 @@ int mbedtls_ssl_tls13_write_early_data_ext(mbedtls_ssl_context *ssl,
 
     return 0;
 }
+
+#if defined(MBEDTLS_SSL_SRV_C)
+int mbedtls_ssl_tls13_check_early_data_len(mbedtls_ssl_context *ssl,
+                                           size_t early_data_len)
+{
+    uint32_t uint32_early_data_len = (uint32_t) early_data_len;
```

Comment:
We can be sure that the size of early data fits into 32 bits, since RFC 8446 says

>      struct {} Empty;
>
>      struct {
>          select (Handshake.msg_type) {
>              case new_session_ticket:   uint32 max_early_data_size;
>              case client_hello:         Empty;
>              case encrypted_extensions: Empty;
>          };
>      } EarlyDataIndication;

Reply:
> Can we be sure that `early_data_len` fits into a `uint32_t` here? Should we check and return an error if it is bigger?

In theory no if SIZE_MAX > UINT32_MAX, in practice very likely as `early_data_len` is related to the size of records or content into records and the size of a record is smaller than 2^16 + 1 for TLS 1.3 if I remember correctly, for sure well below UINT32_MAX. Anyway I will add the check you propose, safer, thanks.

Reply:
Actually, not sure we need the cast anyway. I may have introduce it just for the log which is not a very good reason. I am trying without (changed the log), let's see what the CI says.